### ISSUE TITLE
Admin endpoint - can't set address, only socket path is configured

### ISSUE THREAD

**Issue by Daxcor69** [projectcontour/contour#5082]
Installing contour with the quickstart yamls.  I made the necessary changes for my environment.  I am having issues with too many connections and heap memory.  I am trying to enable the admin endpoint so that I can query the memory endpoint etc.  I have tried everything to make this endpoint available.  

When I look at the contour.yaml in the contour container I see:
```
accesslog-format: envoy
disablePermitInsecure: false
network:
  admin-port: 9001
tls:
  fallback-certificate: {}
```
My understanding is that by specifying the port, this enables the admin endpoint.  When I describe the envoy container there is no port 9001 is listed.  I have tried to port forward the port anyway to see if it was there but not listed.  This didn't work.

When I enter the envoy container I look in the /admin dir and find admin.sock. I am not seeing where it exposes the port on an IP, only a socket on the file system.  When I look in the /config dir  envoy.json I get the following:
```
{
  "static_resources": {
    "clusters": [
      {
        "name": "contour",
        "alt_stat_name": "contour_contour_8001",
        "type": "STRICT_DNS",
        "connect_timeout": "5s",
        "load_assignment": {
          "clu ster_name": "contour",
          "endpoints": [
            {
              "lb_endpoints": [
                {
                  "endpoint": {
                    "address": {
                      "socket_address": {
                        "address": "contour",
                        "port_value": 8001
                      }
                    }
                  }
                }
              ]
            }
          ]
        },
        "circuit_breakers ": {
          "thresholds": [
            {
              "priority": "HIGH",
              "max_connections": 100000,
              "max_pending_requests": 100000,
              "max_requests": 60000000,
              "max_retries": 50
            },
            {
              "max_connections": 100000,
              "max_pending_requests": 100000,
              "max_requests": 60000000,
              "max_retries": 50
            }
          ]
        },
        "typed_extension_protocol_options": {
          "envoy.extensions.upstreams.http.v3.HttpPro tocolOptions": {
            "@type": "type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions",
            "explicit_http_config": {
              "http2_protocol_options": {
                
              }
            }
          }
        },
        " transport_socket": {
          "name": "envoy.transport_sockets.tls",
          "typed_config": {
            "@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsCo ntext",
            "common_tls_context": {
              "tls_params": {
                "tls_maximum_protocol_version": "TLSv1_3"
              },
              "tls_certificate_sds_secret_configs": [
                {
                  "name": "contour_xds_tls_certific ate",
                  "sds_config": {
                    "path_config_source": {
                      "path": "/config/resources/sds/xds-tls-certificate.json"
                    },
                    "resource_api_version": "V3"
                  }
                }
              ],
              "validation_context_sds_sec ret_config": {
                "name": "contour_xds_tls_validation_context",
                "sds_config": {
                  "path_config_source": {
                    "path": "/config/resources/sds/xds-validation-context.json"
                  },
                  "re source_api_version": "V3"
                }
              }
            }
          }
        },
        "upstream_connection_options": {
          "tcp_keepalive": {
            "keepalive_probes": 3,
            "keepalive_time": 30,
            "keepalive_interval": 5
          }
        }
      },
      {
        "name": "en voy-admin",
        "alt_stat_name": "contour_envoy-admin_9001",
        "type": "STATIC",
        "connect_timeout": "0.250s",
        "load_assignment": {
          "cluster_name": "envoy-admin",
          "endpoints": [
            {
              "lb_endpoints": [
                {
                  "endpoint": {
                    "address": {
                      "pipe": {
                        "path": "/admin/admin.sock",
                        "mode": 420
                      }
                    }
                  }
                }
              ]
            }
          ]
        }
      }
    ]
  },
  "dynamic_resources": {
    "lds_config": {
      "api_config_source": {
        "api_type": "GRPC",
        "transport_api_version": "V3",
        "grpc_services": [
          {
            "envoy_grpc": {
              "cluster_name": "contour",
              "authority": "contour"
            }
          }
        ]
      },
      "resource_api_version": "V3 "
    },
    "cds_config": {
      "api_config_source": {
        "api_type": "GRPC",
        "transport_api_version": "V3",
        "grpc_services": [
          {
            "envoy_grpc": {
              "cluster_name": "contour",
              "authority": "c ontour"
            }
          }
        ]
      },
      "resource_api_version": "V3"
    }
  },
  "layered_runtime": {
    "layers": [
      {
        "name": "base",
        "static_layer": {
          "re2.max_program_size.error_level": 1048576,
          "re2.max_pr ogram_size.warn_level": 1000
        }
      },
      {
        "name": "dynamic",
        "rtds_layer": {
          "name": "dynamic",
          "rtds_config": {
            "api_config_source": {
              "api_type": "GRPC",
              "transport_api_version": "V3",
              "grpc_services": [
                {
                  "envoy_grpc": {
                    "cluster_name": "contour",
                    "authority": "contour"
                  }
                }
              ]
            },
            "resource_api_version": "V3"
          }
        }
      },
      {
        "name": "admin",
        "admin_layer": {
          
        }
      }
    ]
  },
  "admin": {
    "access_log": [
      {
        "name": "envoy.access_loggers.file",
        "typed_config": {
          "@type": "type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLo g",
          "path": "/dev/null"
        }
      }
    ],
    "address": {
      "pipe": {
        "path": "/admin/admin.sock",
        "mode": 420
      }
    }
  },
  "overload_manager": {
    "refresh_interval": "0.250s",
    "resource_monitors": [
      {
        " name": "envoy.resource_monitors.fixed_heap",
        "typed_config": {
          "@type": "type.googleapis.com/envoy.extensions.resource_monitors.fixed_heap.v3.FixedHeapConfig",
          "m ax_heap_size_bytes": "2147483648"
        }
      }
    ],
    "actions": [
      {
        "name": "envoy.overload_actions.shrink_heap",
        "triggers": [
          {
            "name": "envoy.resource_monitors.fixed_heap",
            "thresh old": {
              "value": 0.95
            }
          }
        ]
      },
      {
        "name": "envoy.overload_actions.stop_accepting_requests",
        "triggers": [
          {
            "name": "envoy.resource_monitors.fixed_heap",
            "threshold": {
              "value ": 0.98
            }
          }
        ]
      }
    ]
  },
  "default_regex_engine": {
    "name": "envoy.regex_engines.google_re2",
    "typed_config": {
      "@type": "type.googleapis.com/envoy.extensions.regex_engines.v3. GoogleRE2"
    }
  }
}
```

**Comment by Daxcor69** [comment]
In the documentation it says:
```
Bootstrap Flags
There are flags that can be passed to contour bootstrap that help configure how Envoy connects to Contour:

Flag	Default	Description

–admin-address	/admin/admin.sock	Path to Envoy admin unix domain socket.
–overload-max-heap | "" | Defines the maximum heap size in bytes until Envoy overload manager stops accepting new connections.


I am trying to figure out how to change the bootstrap values either with the yams or the helm chart. I have tried ENV and ARG on both contour and envoy with no success.  I am also trying to set the overload-max-heap. I edited the deamonset manually and set both boot strap flags.  It accepted the max-heap but it didn't accept the admin-address.  I am not sure what to  do

**Comment by tsaarni** [comment]
The full admin API is exposed directly only via UNIX domain socket `/admin/admin.sock` for [security reasons](https://github.com/projectcontour/contour/security/advisories/GHSA-5ph6-qq5x-7jwc). It can be accessed with curl's `--unix-socket` option. After Envoy has started and successfully connected to Contour, it will be dynamically configured with a listener `127.0.0.1:<admin-port>` and a [limited set of end-points](https://github.com/projectcontour/contour/blob/d044dbd7b72587f4e40052af13f76439f25ed363/internal/envoy/v3/stats.go#L85-L108). That listener will not show up in the bootstrap config file, but it should work like described in the [troubleshooting page](https://projectcontour.io/docs/v1.24.1/troubleshooting/envoy-admin-interface/). If not, could there be something preventing Envoy from connecting and receiving dynamic config from Contour?

**Comment by Daxcor69** [comment]
ok, so that is weird.  I was trying to port forward to a windows client, and then trying to connect via a chrome browser.  I decided to go to a Linux terminal and do a port forward.  Then I was able to access the endpoints via curl.  So this issue can be closed as the endpoint is working however not on a windows client.  This was very confusing as it didn't show up on the list of ports that could be forwarded when I described the pod. Thank you for the explanation.
